thunder/core/vjp_utils.py

```diff
@@ -178,3 +178,45 @@ def bw_fn(*args, **kwargs):
     _cache[key] = fw_fn, bw_fn
 
     return fw_fn, bw_fn
+
+
+def get_saved_for_backward_tensors(trace: TraceCtx) -> tuple[TensorProxy]:
+    """
+    Given a trace, return the tensors that are saved for backward in the trace.
+
+    Args:
+        trace: The trace to extract saved tensors from.
+
+    Example:
+        >>> import thunder
+        >>> import torch
+        >>> from thunder.core.vjp_utils import get_saved_for_backward_tensors
+        >>> def forward(a, b): return a * b
+        >>> a = torch.tensor(1.0, requires_grad=True)
+        >>> b = torch.tensor(2.0, requires_grad=True)
+        >>> jitted_forward = thunder.jit(forward)
+        >>> jitted_forward(a, b)
+        tensor(2., grad_fn=<ThunderFunctionBackward>)
+        >>> trace = thunder.last_traces(jitted_forward)[-1]
+        >>> saved_tensors = get_saved_for_backward_tensors(trace)
+        >>> print(saved_tensors)
+        (<TensorProxy(name="a", dtype=thunder.dtypes.float32, shape=())>, <TensorProxy(name="b", dtype=thunder.dtypes.float32, shape=())>)
+    """
+    # First let's check if the trace was generated by Thunder's automatic differentiation
+    # The name is defined at
+    # https://github.com/Lightning-AI/lightning-thunder/blob/681aef744786c85f925a590e6b5e6b5162932adf/thunder/core/transforms.py#L3643
+    utils.check(
+        lambda: trace.fn.__name__ == "augmented_forward_fn",
```

Comment:
This should be bool - `trace.fn.__name__ == "augmented_forward_fn"` and the next argument should be a lambda producing string.

Reply:
Thanks for catching this!